packages/commonwealth/client/scripts/controllers/app/login.ts

```diff
@@ -178,15 +187,36 @@ export async function updateActiveAddresses({
   } else if (app.user.activeAccounts.length === 0) {
     // no addresses - preview the community
   } else {
-    const existingAddress = app.roles.getDefaultAddressInCommunity({
-      chain: chain.id,
-    });
+    // Find all addresses in the current community for this account, sorted by last used date/time
+    const communityAddressesSortedByLastUsed =
+      app.roles.getSortedLastUsedAddressInCommunity({
+        chain: chain.id,
+      });
+
+    // From the sorted adddress in the current community, find an address which has an active session key
+    const chainBase = app.chain?.base;
+    const idOrPrefix =
+      chainBase === ChainBase.CosmosSDK
+        ? app.chain?.meta.bech32Prefix
+        : app.chain?.meta.node?.ethChainId;
+    const canvasChainId = chainBaseToCanvasChainId(chainBase, idOrPrefix);
+    const foundAddressWithActiveSessionKey =
+      await communityAddressesSortedByLastUsed.find(async (acc) => {
```

Comment:
I found some sessions signers have `hasAuthenticatedSession` defined as an async method https://github.com/hicommonwealth/commonwealth/blob/1bd8d3f56d7226b08726edbb1795be8df401b017/packages/commonwealth/client/scripts/controllers/server/sessionSigners/cosmos.ts#L31-L40

Also, I am not getting the lint error locally.

cc: @masvelio

Reply:
I think the issue here is the await on `communityAddressesSortedByLastUsed`. Since communityAddressesSortedByLastUsed is an AddressInfo[], we don't need to await it.

But I have a separate comment here. We should use something like:
``` typescript
const foundAddressWithActiveSessionKey = await Promise.race(
  communityAddressesSortedByLastUsed.map(async (acc) => {
    const hasSession = await app.sessions
      .getSessionController(chainBase)
      .hasAuthenticatedSession(canvasChainId, acc.address);

    if (hasSession) {
      return acc;
    }
  })
);
```
I think, because the current code will handle the awaits one at a time, whereas this code will do them in parallel.

Reply:
if the goal is to find the first one to keep the right order, you should probably try a regular for loop

Reply:
> I think the issue here is the await on communityAddressesSortedByLastUsed. Since communityAddressesSortedByLastUsed is an AddressInfo[], we don't need to await it.

actually, we are awaiting on the valid session of an address in `communityAddressesSortedByLastUsed` and we only have to find the first valid account so `Promise.race` won't work there.

> if the goal is to find the first one to keep the right order, you should probably try a regular for loop

we are already using `.find` which gets the first one but if you mean using the `for loop` syntax specifically that can be done.

Reply:
Updated the implementation to use for loop. cc: @Rotorsoft